Add rotation_shutdown

wehimwich opened the pull request:
This is a concept for consideration. The basic idea is to have one type-in that will check if it is safe to do a disk rotation and if is, shut the system down so it can be done. It is part of the quest to reduce the chances that an error will be made at a remote location that could lead to a tedious long-distance recovery (once it is figured out what had gone wrong). I know that sometimes I start a rotation and then wonder if the disks were really synced.

I am not sure what the best name for this script would be. This was the best of the options that I had thought of so far. The script may be useful in some situations where a rotation is not going to be done, but maybe those are rare enough that it is okay that name for the most common use doesn't quite fit.

jfhquick reviewed (changes requested):
This looks sane enough to me.  I think you should call it "safe_shutdown" to avoid people finding out the hard way what it actually does.  I rather expected that it would also check for logged-in users and/or a running FS before shutting down, but then I don't know your use case.

wehimwich commented:
Thanks for your corrections and feedback!

I changed the testing of the RAID state to only depend on `mdadm --detail` and to use the progress meter logic of _refresh_secondary_. I think this may help prevent doing the wrong thing if _/proc/mdstat_ is late in showing progress. This also makes it consistent with the other scripts.

I take your point on the name. The script is really aimed at reducing the risk of an error with the most common operator interaction with the RAID, a rotation. My suggestion now would be _rotation_shutdown_. That is a bit cumbersome to type, but tab completion can help (except for CIS hardened systems). I think using _safe_shutdown_ makes it sound like it is generically useful, but there are cases where it is safe to shutdown with recovering/degraded arrays and it wouldn't work for those. From the existing RAID document, it looks like the only other articulated cases where this script might be useful are for "Recoverable testing", in the middle of "Recover from a shelf disk", and when you have two good disks installed and want to "Initialize a new disk". Maybe it be would worthwhile to include a tip for those cases saying that the _rotation_shutdown_ could be used, if the output messages are interpreted appropriately. Otherwise having versions with appropriate names and messages for these other cases seems unnecessarily complicated; who would remember to use them.

I added a check to inhibit shutdown if the FS is running.  Perhaps it should also check for _autoftp_ and _fs.prompt_. Checking for other users is a little more fraught. If _sudo_ has been used to promote to _root_ that will create another user, but that case could probably be identified with the output of _w_. Maybe there is no need for any of this if the usage is limited to shutting down for rotations (and the few rare other cases). The check for the FS is easy. The other checks could be added, if desired.
